src/connector/src/parser/mod.rs

```diff
@@ -872,7 +872,11 @@ pub enum ProtocolProperties {
 
 impl SpecificParserConfig {
     // The validity of (format, encode) is ensured by `extract_format_encode`
-    pub fn new(info: &StreamSourceInfo, props: &HashMap<String, String>) -> Result<Self> {
+    pub fn new(
+        info: &StreamSourceInfo,
+        props: &HashMap<String, String>,
+        row_options: Option<&mut BTreeMap<String, String>>,
```

Comment:
why wrap with an Option? I think we can always get the btreemap from the frontend